index.js

```diff
@@ -18,30 +19,48 @@ var startTime = function() {
   });
 };
 
+var potentiallyMigrateDatabase = function() {
+  return new Promise(function (resolve, reject) {
+    postgrator.setConfig({
+      migrationDirectory: __dirname + '/migrations',
+      driver: 'pg',
+      connectionString: process.env.POSTGRESQL_URL
+    });
+
+    postgrator.migrate('201505271423', function(err, migrations) {
+      if (err) {
+        reject(err);
+      } else {
+        postgrator.endConnection(function() {
+          resolve(migrations);
+        });
+      }
+    });
+  });
+};
+
 var loop = function () {
   var starttime;
   startTime()
     .then(function (time) {
       starttime = time;
     })
-
+    .then(potentiallyMigrateDatabase)
+    .then(function(migrations) {
```

Comment:
I don't understand where the return value comes out of potentiallyMigrateDatabase.

I see a single resolve under the '201505271423' timestamp.

How does this work if there were more migrations (each with a resolve)? It is just a list that gets built from concatenating the result of thenables, with each migration as a following then?